src/common/viz/FigureExtractor.js

```diff
@@ -0,0 +1,145 @@
+            type: 'subgraph',
+            graphId: this._client.getNode(graphId).getAttribute('id'),
+            subgraphId: node.getAttribute('id'),
+            subgraphName: node.getAttribute('name'),
+            title: node.getAttribute('title'),
+            xlim: node.getAttribute('xlim'),
+            ylim: node.getAttribute('ylim'),
+            xlabel: node.getAttribute('xlabel'),
+            ylabel: node.getAttribute('ylabel'),
+        };
+
+        const children = node.getChildrenIds().map(id => this._client.getNode(id));
+        desc.lines = children.filter(node => this.getMetaType(node) === EXTRACTORS.LINE)
+            .map(lineNode => this[this.getMetaType(lineNode)](lineNode));
+        desc.images = children.filter(node => this.getMetaType(node) === EXTRACTORS.IMAGE)
+            .map(imageNode => this[this.getMetaType(imageNode)](imageNode));
+
+        return desc;
+    };
+
+    FigureExtractor.prototype[EXTRACTORS.LINE] = function (node) {
+        const ids= this.getIdHierarchy(node, 5),
+            id = ids[0],
+            execId = ids[ids.length-1];
+        let points, desc;
+
+        points = node.getAttribute('points').split(';')
+            .filter(data => !!data)  // remove any ''
+            .map(pair => {
+                const [x, y] = pair.split(',').map(num => parseFloat(num));
+                return {x, y};
+            });
+        desc = {
+            id: id,
+            execId: execId,
+            subgraphId: this._client.getNode(node.getParentId()).getAttribute('id'),
+            lineName: node.getAttribute('name'),
+            label: node.getAttribute('label'),
+            name: node.getAttribute('name'),
+            type: 'line',
+            points: points,
+            color: node.getAttribute('color')
+        };
+
+        return desc;
+    };
+
+    FigureExtractor.prototype[EXTRACTORS.IMAGE] = function (node) {
+        const ids= this.getIdHierarchy(node, 5),
+            id = ids[0],
+            execId = ids[ids.length-1],
+            imageHeight = node.getAttribute('height'),
+            imageWidth = node.getAttribute('width'),
+            numChannels = node.getAttribute('numChannels');
+        const colorModel = numChannels === 3 ? 'rgb' : 'rgba';
+        return {
+            id: id,
+            execId: execId,
+            subgraphId: this._client.getNode(node.getParentId()).getAttribute('id'),
+            type: 'image',
+            width: imageWidth,
+            height: imageHeight,
+            colorModel: colorModel,
+            visible: node.getAttribute('visible'),
+            rgbaMatrix: Utils.base64ToImageArray(node.getAttribute('rgbaMatrix'), imageWidth, imageHeight, numChannels)
+        };
+    };
+
+    FigureExtractor.prototype.compareSubgraphIDs = function (desc1, desc2) {
+        if (desc1.subgraphId >= desc2.subgraphId) return 1;
+        else return -1;
+    };
+
+    FigureExtractor.prototype.getIdHierarchy = function (node, level) {
```

Comment:
There may be a better name for this. Maybe something like `getContainmentAncestorIds` since it walks the path from the given node towards the root node in the containment tree `level` steps.